Mathlib/Data/List/Defs.lean

```diff
@@ -749,4 +749,72 @@ Example: if `f : ℕ → list ℕ → β`, `List.mapWithComplement f [1, 2, 3]`
 def mapWithComplement {α β} (f : α → List α → β) : List α → List β :=
   mapWithPrefixSuffix fun pref a suff => f a (pref ++ suff)
 
+/--
+Auxiliary function for `List.compareWith`
+-/
+def compareWithAux [Ord α] : List α → List α → Ordering
+  | x::xs, y::ys =>
+    let c := compare x y
+    if c == Ordering.eq then
+      compareWithAux xs ys
+    else c
+  | [], [] => Ordering.eq
+  | _, [] => Ordering.gt
+  | [], _ => Ordering.lt
+
+/--
+`List.compareWith` compares two lists `l₁` and `l₂` using thee function `compare` from
+the `Ord` typeclass, which means that the type of the list's elements have to have
+an instance of this typeclass. `List.comparewith` returns a contructor from the
+`Ordering` inductive type.
+
+Examples:
+```
+[1, 5].compareWith compare' [1, 8] = Ordering.lt
+[1, 10].compareWith compare' [1, 10] = Ordering.eq
+[1, 11].compareWith compare' [1, 13] = Ordering.gt
+[1, 5].compareWith compare' [1] = Ordering.lt
+[1].compareWith compare' [1, 8] = Ordering.gt
+```
+-/
+def compareWith [Ord α] (l₁ l₂ : List α) : Ordering :=
```

Comment:
What I was suggesting was this:
```suggestion
def compareWith (f : α → α → Ordering) (l₁ l₂ : List α) : Ordering :=
```

But this version is also acceptable.

Reply:
I think just using the built-in `compare` function is fine, right?

Reply:
I also fixed the docstrings

Reply:
Maybe I should make a second function that takes a custom function.